python/python/lance/dataset.py

```diff
@@ -175,6 +176,10 @@ def scanner(
             number of rows (or be empty) if the rows closest to the query do not
             match the filter.  It's generally good when the filter is not very
             selective.
+        with_row_id: bool, default False
+            If set true, a column named ``_row_id`` will be added to the result, which
+            represents the internal id of the each row.
+
```

Comment:
This is internal use only right? Perhaps we shouldn't document this option publicly?

Reply:
Ok, fair.